indra/newview/lluilistener.cpp

```diff
@@ -264,6 +266,13 @@ LLMenuGL* get_parent_menu(LLEventAPI::Response& response, const LLSD&event)
     return parent_menu;
 }
 
+// Return event["pos"].asInteger() if passed, but clamp (0 <= pos <= size).
+// Reserve -1 return to mean event has no "pos" key.
+S32 get_pos(const LLSD& event, U32 size)
+{
+    return event["pos"].isInteger()? llclamp(event["pos"].asInteger(), 0, size) : -1;
```

Comment:
0 menu position is used for 'double line' in the top menu (which allows to undock that menu). Should we allow inserting new items before it, basically pushing 'double line' down? It's ok for me if you think we should (though it's somewhat inconsistent), Otherwise, let's clamp to 1.

Reply:
Hmm! I based this `llclamp()` call on the test in `LLMenuGL::insert()`. Are you saying that if we allow inserting at 0, tearing off the menu would leave that first (inserted) item attached to the top menu?

If we change this to 1, should we also change the test in `LLMenuGL::insert()`?

Reply:
Using test_top_menu.lua: current script vs inserting "Debug console" at 0.
![image](https://github.com/user-attachments/assets/ade08496-81a4-4cd5-b119-c3fdd0ef9432)

I'm not sure about `LLMenuGL::insert()`. As far as I remember, only top menus have these 'tear-off' items(LLMenuItemTearOffGL). So, it makes sense to allow inserting at 0 in general for other menus, the exception is the top menu.